### ISSUE TITLE
[FORMS] Form calls ->removeValidation() on validator, but may not always be defined.

### ISSUE THREAD

**Issue by patricknelson** [silverstripe/silverstripe-framework#4417]
On this line: https://github.com/silverstripe/silverstripe-framework/blame/3.2/forms/Form.php#L683

``` php
        // We have to remove validation, if the fields are not editable ;-)
        if($this->validator)
            $this->validator->removeValidation();
```

However, if you create your own implementation of `Validator` there's no contract (abstract definition) to inform you that you must also implement a `->removeValidation()` method on your validator. I'd suggest either defining this as a default method on the `Validator` abstract class or, better yet, set this up as an abstract method so that it will fail earlier so that results are more consistent and predictable (since you may not realize this will fail until it's too late and already in production).

Thankfully I haven't had taht issue just yet, but I didn't realize this method was needed until months after I ended up disabling all the fields on my `DataObject` :)

**Comment by patricknelson** [comment]
FYI, this affects `3.1.x` and `3.2` from what I can tell.

**Comment by sminnee** [comment]
It looks like this was fixed in #4542 which has been available since 4.0; closing
